### PR TITLE
Documentation: Installing with Doom Emacs

### PR DESCRIPTION
**Pull request by teesloane**
###### Motivation for this change

Partial contribution to https://github.com/jethrokuan/org-roam/issues/209: Add notes on installing `org-roam` with Doom Emacs.

---

Let me know if this is of use / needs work. Thank you for your hard work on this package!

### CODE REVIEW

File: doc/installation.md [hunk 1]
```diff
@@ -88,3 +88,28 @@ If you are using Spacemacs, you can easily install org-roam by creating a simple
...
+
+```elisp
+;; ~/.doom.d/config.el
+(use-package! org-roam
+  :hook (after-init . org-roam-mode)
+  :config
+  (setq-default
+   org-roam-directory "~/<path>/<to>/<roam_directory>"
+   org-roam-link-title-format "%sº" ;; appends a  `º` to each Roam link.
+  (org-roam-build-cache))
```
Comment (jethrokuan) [inline doc/installation.md]: Not needed anymore

File: doc/installation.md [hunk 2]
```diff
@@ -88,3 +88,28 @@ If you are using Spacemacs, you can easily install org-roam by creating a simple
...
+Subsequently, in your `~/.doom.d/config.el` file, configure org-roam:
+
+```elisp
+;; ~/.doom.d/config.el
+(use-package! org-roam
+  :hook (after-init . org-roam-mode)
+  :config
+  (setq-default
+   org-roam-directory "~/<path>/<to>/<roam_directory>"
+   org-roam-link-title-format "%sº" ;; appends a  `º` to each Roam link.
```
Comment (jethrokuan) [inline doc/installation.md]: Maybe leave this out, seems like personal preference. Also there's a less invasive option of setting the font face.

File: doc/installation.md [hunk 3]
```diff
@@ -88,3 +88,28 @@ If you are using Spacemacs, you can easily install org-roam by creating a simple
...
+
+If you are using [Doom Emacs](https://github.com/hlissner/doom-emacs), configure packages as explained in the [getting started](https://github.com/hlissner/doom-emacs/blob/develop/docs/getting_started.org#configuring-packages) guide. 
+
+Declare org-roam as a package in your `~/.doom.d/packages.el`:
+
+```elisp
+;; ~/.doom.d/packages.el
+
+(package! org-roam
+  :recipe (:host github :repo "jethrokuan/org-roam" :branch "develop"))
```
Comment (asymmetric) [inline doc/installation.md]: Is the `:branch` necessary, if `develop` is the default branch?

File: doc/installation.md (hunk 1 above)
Comment (asymmetric) [reply doc/installation.md]: Do we need `org-roam--build-cache-async` instead?

File: doc/installation.md [hunk 4]
```diff
@@ -88,3 +88,28 @@ If you are using Spacemacs, you can easily install org-roam by creating a simple
...
+```
+
+Subsequently, in your `~/.doom.d/config.el` file, configure org-roam:
+
+```elisp
+;; ~/.doom.d/config.el
+(use-package! org-roam
```
Comment (asymmetric) [inline doc/installation.md]: Do we need a `:after org` here? 

/cc @hlissner

File: doc/installation.md (hunk 3 above)
Comment (jethrokuan) [reply doc/installation.md]: for now yes, I'll probably change it again when I push out 1.0.0.

File: doc/installation.md (hunk 1 above)
Comment (jethrokuan) [reply doc/installation.md]: That function no longer exists in `develop`.

File: doc/installation.md (hunk 4 above)
Comment (jethrokuan) [reply doc/installation.md]: No. If you wish to have the bindings available before Org is loaded, then you can leave out `:after org`.

**Comment by teesloane** [comment]
> Are there no sane bindings? Don't use modal editing, so not sure how to do it.

It depends, I think. Doom manages a lot of it's bindings in a single place using the `(map! ...)` feature. I tend to mimic that, customizing my own block of bindings in a single place in my config. With that said, I've moved to add a "map!" usage into the use-package! declaration. It's much clearer anyway, so I'll prob start doing that in the future.

**Comment by jethrokuan** [comment]
> I guess bindings can clash too, so maybe it's worth mentioning that?

Yeah probably is worth mentioning!

**Comment by bdarcus** [comment]
I'm not sure if this is ideal, but this is what I have for the bindings.

```elisp
(map!
 :map (org-mode-map)
 :localleader
 (:prefix ("R" . "org-roam")
   "r" #'org-roam
   "f" #'org-roam-find-file
   "i" #'org-roam-insert
   "g" #'org-roam-show-graph))
```

**Comment by teesloane** [comment]
@bdarcus That looks nice too -- I didn't know you could use the `:map` symbol like that. I grouped my bindings in with `notes` so that I could easily jump to any note from any mode ¯\_(ツ)_/¯

**Comment by jethrokuan** [comment]
> I'm not sure if this is ideal, but this is what I have for the bindings.
> 
> ```emacs-lisp
> (map!
>  :map (org-mode-map)
>  :localleader
>  (:prefix ("R" . "org-roam")
>    "r" #'org-roam
>    "f" #'org-roam-find-file
>    "i" #'org-roam-insert
>    "g" #'org-roam-show-graph))
> ```

Doesn't this mean that the functions are only available in org-mode? The best is to bind them to the global `org-roam-mode`.

File: doc/installation.md [hunk 5]
```diff
@@ -88,3 +88,34 @@ If you are using Spacemacs, you can easily install org-roam by creating a simple
...
+  :hook (after-init . org-roam-mode)
+  :config
+  (setq-default
+   org-roam-directory "/path/to/org-files/")
+  ;; Note bindings could clash / overwrite Doom Bindings. 
+  (map!
+      (:leader
+        (:prefix-map ("n" . "notes")
```
Comment (hlissner) [inline doc/installation.md]: I suggest `:prefix "n"` instead. `:prefix-map` shouldn't be used directly (it's purpose to define a prefix keymap, which isn't useful for the end user to do).

Also, no need to label a prefix that already has a label.

File: doc/installation.md (hunk 4 above)
Comment (hlissner) [reply doc/installation.md]: If we're binding leader keys, we could bind them early and activate org-roam-mode when any of those commands autoload the package. i.e.

```elisp
(use-package! org-roam
  :commands (org-roam-insert org-roam-find-file org-roam)
  :init 
  (setq org-roam-directory "/path/to/org-files/")
  (map! :leader 
        :prefix "n"
        :desc "Org-Roam-Insert" "i" #'org-roam-insert
        :desc "Org-Roam-Find"   "/" #'org-roam-find-file
        :desc "Org-Roam-Buffer" "r" #'org-roam)
  :config
  (org-roam-mode +1))
```

What do you think?

**Pull request merged**